### PR TITLE
feat: dataspace protocol transfer-process-dispatcher|spi

### PR DESCRIPTION
**Pull request by janpmeyer**
## What this PR changes/adds

Implements transfer-process spi and dispatcher following :dsp:dsp-catalog.

`:data-protocols:dsp:dsp-transfer-process:dsp-transfer-process-dispatcher`
`:data-protocols:dsp:dsp-transfer-process:dsp-transfer-process-spi`

## Why it does that

Support the dataspace protocol specification.

## Further notes

Waits for PR #2759 to be merged

## Linked Issue(s)

Relates #2474

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [x] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

**Review by jimmarino — CHANGES_REQUESTED** [review]
These delegates and tests have a lot of duplicate code. Please consolidate them per comments on the other PRs and we will revisit this PR

**Review by ndr-brt — APPROVED** [review]
I think there's some refactoring to do on the `TestBase` thing. Why tests there are not annotated with `@Test`? Doing like this there will no need to recreate tests on the subclass.
Let's do that in a separate PR.

File: data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-http-dispatcher/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/dispatcher/DspTransferProcessDispatcherExtension.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+/**
+ *
+ * Provides HTTP dispatching for Dataspace Protocol transfer process messages via the {@link DspHttpRemoteMessageDispatcher}.
+ */
+@Extension(value = DspTransferProcessDispatcherExtension.NAME)
+public class DspTransferProcessDispatcherExtension implements ServiceExtension {
+
+    public static final String NAME = "Dataspace Protocol: Control Plane Dispatcher Extension";
```
Comment (jimmarino) [inline data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-http-dispatcher/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/dispatcher/DspTransferProcessDispatcherExtension.java:41]: For naming, please follow the conventions of other extensions, keeping the name short and devoid of colons.

**Review by jimmarino — CHANGES_REQUESTED** [review]
one small change

File: data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-http-dispatcher/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/dispatcher/DspTransferProcessDispatcherExtension.java (hunk 1 above)
Comment (janpmeyer) [reply data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-http-dispatcher/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/dispatcher/DspTransferProcessDispatcherExtension.java]: Orientated on Catalog and Negotiation Dispatcher. Hope this Name is fine.

**Pull request merged**